mongo/datadog_checks/mongo/dbm/operation_samples.py

```diff
@@ -0,0 +1,347 @@
+                )
+
+                explain_plan = self._get_explain_plan(op=operation.get("op"), command=command)
+                sample = self._create_operation_sample_payload(
+                    now, operation_metadata, obfuscated_command, query_signature, explain_plan, activity
+                )
+                yield activity, sample
+            except Exception as e:
+                self._check.log.error("Unexpected error while collecting operation samples: %s", e)
+                continue
+
+    def _get_current_op(self):
+        operations = self._check.api_client.current_op()
+        for operation in operations:
+            self._check.log.debug("Found operation: %s", operation)
+            yield operation
+
+    def _should_include_operation(self, operation: dict) -> bool:
+        command = operation.get("command")
+        if not command:
+            self._check.log.debug("Skipping operation without command: %s", operation)
+            return False
+        if "hello" in command and operation.get("op") == "command":
+            # MongoDB drivers and clients use hello to determine the state of
+            # the replica set members and to discover additional members of a replica set.
+            self._check.log.debug("Skipping hello operation: %s", operation)
+            return False
+        return True
+
+    def _should_explain(self, op: Optional[str], command: dict) -> bool:
+        dbname = command.get("$db")
+        if not dbname or dbname == "admin":
+            # Skip system operations
+            self._check.log.debug("Skipping explain system operation: %s", command)
+            return False
+
+        if not op or op == "none":
+            # Skip operations that are not queries
+            self._check.log.debug("Skipping explain operation without operation type: %s", command)
+            return False
+
+        if op in ("getMore", "insert", "update", "getmore", "killcursors", "remove"):
+            # Skip operations that are not queries
+            self._check.log.debug("Skipping explain operation type %s: %s", op, command)
+            return False
+
+        return True
+
+    def _get_explain_plan(self, op: Optional[str], command: dict) -> OperationSamplePlan:
+        if not self._should_explain(op, command):
+            return
+
+        dbname = command.pop("$db")
+        try:
+            explain_plan = self._check.api_client[dbname].command("explain", command)
+            if "command" in explain_plan:
+                explain_plan.pop("command")  # Remove the original command from the explain plan
+            query_planner = self._get_query_planner(explain_plan)
+            return {
+                "definition": explain_plan,
+                "query_hash": query_planner.get("queryHash"),
+                "plan_cache_key": query_planner.get("planCacheKey"),
+                "signature": compute_exec_plan_signature(json_util.dumps(explain_plan)),
```

Comment:
fine to keep this here, but we should consider generating explain plan signatures in the backend in case we want to tweak it. the current signature generation is not great. it creates separate signatures for essentially the same plan with marginally different values for some of the operations: https://app.datadoghq.com/databases/samples?query=dbm_type%3Aplan%20%40db.query_signature%3A2942df0b83a02a89%20&agg_m=count&agg_m_source=base&agg_t=count&cols=&dbSampleDetailTab=eventAttributes&dbType=MySQL&fromUser=false&viz=stream_issues&start=1716295459699&end=1716299059699&paused=true

Reply:
good call out. we don't do additional explain plan process for mongo in the backend for now. will leave the signature here. we can do backend calculation when we start tweaking mongo explain plans

Reply:
just FWIW, we currently don't have code to handle this in our backend for any other dbms type. We would need to just quickly decide on how we want to do it so its consistent enough